ksql-common/src/main/java/io/confluent/ksql/function/AggregateFunctionInitArguments.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.function;
+
+import io.confluent.ksql.util.KsqlException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Represents a list of initial arguments for the creation of a
+ * {@link io.confluent.ksql.function.KsqlAggregateFunction}
+ *
+ * <p>The initial arguments are always constants.
+ */
+public class AggregateFunctionInitArguments {
+
+  private final int udafIndex;
+  private final List<String> initArgs;
+
+  public static final AggregateFunctionInitArguments EMPTY_ARGS =
+      new AggregateFunctionInitArguments();
+
+  private AggregateFunctionInitArguments() {
+    this.udafIndex = 0;
+    this.initArgs = Collections.emptyList();
+  }
+
+  public static AggregateFunctionInitArguments ofFunctionArgs(final int index,
+      final List<String> initArgs) {
+    return new AggregateFunctionInitArguments(index, Objects.requireNonNull(
+        initArgs.size() < 2 ? Collections.emptyList() : initArgs.subList(1, initArgs.size())
```

Comment:
this seems a little misleading - If I understand correctly we're using sublist[1:size] because the first argument is the variable and the second+ is the optional constant, but it's tough to get that from just reading the code. I think a comment explaining it at the minimum would be nice or just have the caller handle the input sanitization

Reply:
done